presto-spi/src/main/java/com/facebook/presto/spi/function/SqlInvokedFunction.java

```diff
@@ -98,10 +121,19 @@ public SqlInvokedFunction(
                 .map(Parameter::getType)
                 .collect(collectingAndThen(toList(), Collections::unmodifiableList));
 
-        this.signature = new Signature(functionName, SCALAR, typeVariableConstraints, emptyList(), returnType, argumentTypes, false);
+        this.signature = new Signature(functionName, kind, typeVariableConstraints, emptyList(), returnType, argumentTypes, false);
         this.functionId = new SqlFunctionId(functionName, argumentTypes);
         this.functionVersion = requireNonNull(version, "version is null");
         this.functionHandle = version.hasVersion() ? Optional.of(new SqlFunctionHandle(this.functionId, version.toString())) : Optional.empty();
+
+        if (kind == AGGREGATE && (aggregationMetadata == null || !aggregationMetadata.isPresent())) {
+            throw new IllegalStateException("Aggregate functions require aggregationMetadata");
+        }
+        else if (kind == SCALAR && aggregationMetadata != null && aggregationMetadata.isPresent()) {
```

Comment:
check for any non-aggregate in case support is added for window functions or something.

Reply:
Will update. Thanks! :)